include/pmacc/memory/boxes/DataBox.hpp

```diff
@@ -23,135 +23,55 @@
 #pragma once
 
 #include "pmacc/dimensions/DataSpace.hpp"
-#include "pmacc/types.hpp"
 
 namespace pmacc
 {
-    namespace private_Box
+    namespace detail
     {
-        template<unsigned DIM, class Base>
-        class Box;
-
-        template<class Base>
-        class Box<DIM1, Base> : public Base
+        template<typename DataBox>
+        HDINLINE decltype(auto) access(const DataBox& db, DataSpace<1> idx = {})
         {
-        public:
-            enum
-            {
-                Dim = DIM1
-            };
-            using ValueType = typename Base::ValueType;
-            using RefValueType = typename Base::RefValueType;
-
-            HDINLINE RefValueType operator()(const DataSpace<DIM1>& idx = DataSpace<DIM1>()) const
-            {
-                return Base::operator[](idx.x());
-            }
-
-            HDINLINE RefValueType operator()(const DataSpace<DIM1>& idx = DataSpace<DIM1>())
-            {
-                return Base::operator[](idx.x());
-            }
-
-            HDINLINE Box(Base base) : Base(base)
-            {
-            }
-
-            HDINLINE Box() : Base()
-            {
-            }
-        };
+            return db[idx.x()];
+        }
 
-        template<class Base>
-        class Box<DIM2, Base> : public Base
+        template<typename DataBox>
+        HDINLINE decltype(auto) access(const DataBox& db, DataSpace<2> idx = {})
         {
-        public:
-            enum
-            {
-                Dim = DIM2
-            };
-            using ValueType = typename Base::ValueType;
-            using RefValueType = typename Base::RefValueType;
-
-            HDINLINE RefValueType operator()(const DataSpace<DIM2>& idx = DataSpace<DIM2>()) const
-            {
-                return (Base::operator[](idx.y()))[idx.x()];
-            }
-
-            HDINLINE RefValueType operator()(const DataSpace<DIM2>& idx = DataSpace<DIM2>())
-            {
-                return (Base::operator[](idx.y()))[idx.x()];
-            }
-
-            HDINLINE Box(Base base) : Base(base)
-            {
-            }
-
-            HDINLINE Box() : Base()
-            {
-            }
-        };
+            return db[idx.y()][idx.x()];
+        }
 
-        template<class Base>
-        class Box<DIM3, Base> : public Base
+        template<typename DataBox>
+        HDINLINE decltype(auto) access(const DataBox& db, DataSpace<3> idx = {})
```

Comment:
It should not really matter since `DataSpace<3>` is just 3 integers which will probably be passed in registers. Also the call is forcefully inlined and picongpu is a single TU, so the compiler has full knowledge of all call sites, so it cannot matter :) But ok, I assume it does not hurt either.